str/trans-eqtl/trans_associatr_runner.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-function-docstring,no-member,too-many-arguments
+"""
+This script runs associaTR, given a chromosome or cell type.
+Ensure prior scripts have been run to generate dependent files, particularly:
+- get_cis_numpy_files.py
+- pseudobulk.py
+- get_covariates.py
+- qc_filters_associatr.py (depends on qc_annotator.py)
+
+ analysis-runner --dataset "bioheart" --config trans_associatr_runner.toml \
+    --description "run associatr" \
+    --access-level "test" \
+    --memory "8G" \
+    --output-dir "str/associatr/trans_pilot/bioheart_n975" \
+     python3 trans_associatr_runner.py
+
+
+"""
+import json
+
+import pandas as pd
+
+import hailtop.batch as hb
+
+from cpg_utils import to_path
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import get_batch, init_batch, output_path
+
+
+def main():
+    """
+    Run associaTR processing pipeline
+    """
+    b = get_batch(name='Run associatr')
+    init_batch()
+
+    # Setup MAX concurrency by genes
+    _dependent_jobs: list[hb.batch.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.batch.job.Job):
+        """
+        To avoid having too many jobs running at once, we have to limit concurrency.
+        """
+        if len(_dependent_jobs) >= get_config()['associatr']['max_parallel_jobs']:
+            job.depends_on(_dependent_jobs[-get_config()['associatr']['max_parallel_jobs']])
+        _dependent_jobs.append(job)
+
+    for celltype in get_config()['associatr']['celltypes'].split(','):
+        for chromosome in get_config()['associatr']['chromosomes'].split(','):
+            vcf_file_path = get_config()['associatr']['vcf_file_path']
+            variant_vcf = b.read_input_group(
+                **dict(
+                    base=vcf_file_path,
+                ),
+            )
```

Comment:
this can be moved outside the loops, the VCF isn't specific to celltypes or chromosomes